include/private/rdycoreimpl.h

```diff
@@ -141,12 +148,15 @@ struct _p_RDy {
 
   // solution vectors (global and local)
   Vec X, X_local;
+  Vec Soln;
```

Comment:
What's the difference between `X` and `Soln`? Is one "naturally" ordered and the other isn't?

Reply:
We have been using carelessly using `X` as a temporary Vec (e.g., [src/write_xdmf_output.c#L49](https://github.com/RDycore/RDycore/blob/main/src/write_xdmf_output.c#L49)). May be we need better names for these Vecs.

Reply:
I forgot to mention in my last comment that `X_local` actually holds the solution for owned + ghost cells. Thus, it would be more accurate to rename it as `Soln_local`.